socks/socks5.py

```diff
@@ -170,6 +173,10 @@ def __init__(self) -> None:
         self._received_data = bytearray()
         self._state = SOCKS5State.CLIENT_AUTH_REQUIRED
 
+    @property
+    def state(self) -> SOCKS5State:
+        return self._state
```

Comment:
I'm not 100% about exposing the internal state of the connection to the user, I think it might be useful.